### PR TITLE
✨ Implement Fees ✨

### PR DESCRIPTION
**Pull request by JaredBorders**
Implement Fee Structure

## Description
* Create settings to manage fee(s)
* Implement fee calculations
* Test fee implementation

## Related issue(s)
Closes [✨ Implement Fees ✨](https://github.com/Kwenta/margin-manager/issues/11)

## Not covered
* Gas Optimizations (will do in a separate ticket once limit orders and fees are merged)
* Batch Account Deposit & Open Position into one tx (will do in a separate ticket once limit orders and fees are merged)

## Motivation and Context
💰

### CODE REVIEW

File: contracts/MarginBase.sol [hunk 1]
```diff
@@ -200,6 +202,12 @@ contract MarginBase is MinimalProxyable {
         emit Withdraw(msg.sender, _amount);
     }
 
+    /*///////////////////////////////////////////////////////////////
+                    Account Deposit & Open Position
```
Comment (JChiaramonte7) [inline contracts/MarginBase.sol:206]: Wondering if we should be doing these section type labels for singular functions like this. Might make more sense to break down the section labels by product moving forward. ie. Account Management, Cross Margin, Advanced Orders

File: contracts/MarginBaseSettings.sol [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    /// @param _treasury: Kwenta's Treasury Address
+    /// @param _distributionFee: fee denoted in BPS
+    /// @param _limitOrderFee: fee denoted in BPS
+    /// @param _stopLossFee: fee denoted in BPS
+    constructor(
+        address _treasury,
+        uint256 _distributionFee,
+        uint256 _limitOrderFee,
+        uint256 _stopLossFee
+    ) {
```
Comment (JChiaramonte7) [inline contracts/MarginBaseSettings.sol:90]: don't forget to set the owner

File: contracts/MarginBase.sol (hunk 1 above)
Comment (JaredBorders) [reply contracts/MarginBase.sol:206]: Yeah, it is debatable. I really just followed suit from rari-capital's contracts. I liked the organization by functionality vs the typical way.

File: contracts/MarginBase.sol [hunk 3]
```diff
@@ -217,6 +225,9 @@ contract MarginBase is MinimalProxyable {
             revert MaxNewPositionsExceeded(_newPositions.length);
         }
 
+        /// @notice set totalMarginDelta to 0 and begin tracking fees for current call to distributeMargin()
+        totalMarginDelta = 0;
+
```
[the remark this answers is not in the document]
Comment (JaredBorders) [reply contracts/MarginBase.sol:230]: Absolutely. Was thinking of doing something like that during gas optimizations but I will go ahead and do that now

**Review by JChiaramonte7 — APPROVED** [review]

**Pull request merged**